### PR TITLE
manifest: sdk-connectedhomeip: Update revision

### PR DESCRIPTION
**Pull request by LuDuda**
This commit updates the revision of Matter SDK as well as clarify how to enable CC3xx support for samples with PSA.

### CODE REVIEW

File: doc/nrf/protocols/matter/end_product/security.rst [hunk 1]
```diff
@@ -121,12 +121,12 @@ This is a reference configuration that can be modified in the production firmwar
      - Trusted Storage backend
      - SHA-256 hash
 
-.. [1] The CryptoCell backend is used in parallel with the Oberon backend.
-       It is only used to implement Random Nuber Generation (RNG), and the AED keys derivation driver (only for Thread networking).
-       For all other cryptographic operations, the Oberon backend is used.
+.. [1] The CryptoCell backend is used in parallel with the Oberon backend. By default, the CryptoCell backend is used only for Random Number Generation (RNG) and the AEAD key derivation driver.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/security.rst:124]:
```suggestion
.. [1] The CryptoCell backend is used in parallel with the Oberon backend.
       By default, the CryptoCell backend is used only for Random Number Generation (RNG) and the AEAD key derivation driver.
```

File: doc/nrf/protocols/matter/end_product/security.rst [hunk 2]
```diff
@@ -121,12 +121,12 @@ This is a reference configuration that can be modified in the production firmwar
...
+       To enable the CryptoCell backend for additional operations, set the :kconfig:option:CONFIG_PSA_CRYPTO_DRIVER_CC3XX Kconfig option to true.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/security.rst:125]:
```suggestion
       To enable the CryptoCell backend for additional operations, set the :kconfig:option:`CONFIG_PSA_CRYPTO_DRIVER_CC3XX` Kconfig option to true.
```

File: doc/nrf/protocols/matter/end_product/security.rst [hunk 3]
```diff
@@ -121,12 +121,12 @@ This is a reference configuration that can be modified in the production firmwar
...
 
-.. [2] When the CRACEN driver is used in parallel with the Oberon driver, you need to disable specific CRACEN cryptographic operations to use the Oberon ones.
-       This is because the CRACEN driver has priority when both are enabled.
+.. [2] The CRACEN backend is used in parallel with the Oberon backend. The CRACEN driver is used by default for any supported cryptographic operations.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/security.rst:127]:
```suggestion
.. [2] The CRACEN backend is used in parallel with the Oberon backend.
       The CRACEN driver is used by default for any supported cryptographic operations.
```

File: doc/nrf/protocols/matter/end_product/security.rst [hunk 4]
```diff
@@ -121,12 +121,12 @@ This is a reference configuration that can be modified in the production firmwar
...
+       For all other operations not supported by CRACEN, the Oberon backend is used. If you prefer to use the Oberon implementation for specific cryptographic operations supported by both drivers,
+       you will need to disable those operations in the CRACEN driver, as it takes priority when both are enabled.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/end_product/security.rst:129]:
```suggestion
       For all other operations not supported by CRACEN, the Oberon backend is used.
       To use the Oberon implementation for specific cryptographic operations supported by both drivers, disable those operations in the CRACEN driver, as it takes priority when both are enabled.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
The entries switch between "driver" and "backend" when referring to CRACEN and Oberon, and I'm not sure that's correct. Please unify wording if necessary.

File: subsys/net/openthread/Kconfig [hunk 5]
```diff
@@ -33,7 +33,7 @@ endchoice
 config OPENTHREAD_NRF_SECURITY_PSA
 	bool
 	select NRF_SECURITY
-	select MBEDTLS_ENABLE_HEAP if !OPENTHREAD_COPROCESSOR_RCP
+	select MBEDTLS_ENABLE_HEAP if (!OPENTHREAD_COPROCESSOR_RCP && !BUILD_WITH_TFM)
```
Comment (ArekBalysNordic) [inline subsys/net/openthread/Kconfig:36]:
```suggestion
	select MBEDTLS_ENABLE_HEAP if (!OPENTHREAD_COPROCESSOR_RCP && !BUILD_WITH_TFM && !PSA_SSF_CRYPTO_CLIENT)
```

To cover nRF54H20?

**Review by ArekBalysNordic — APPROVED** [review]

**Review by doublemis1 — APPROVED** [review]
LGTM!

**Review by wiba-nordic — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**